docs/aws/README.md

```diff
@@ -14,33 +14,53 @@ This command will create an IAM Policy with access to all of the resources for a
 ```
 aws iam create-policy --policy-name Karpenter --policy-document "$(cat <<-EOM
 {
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": [
-                "eks:DescribeNodegroup",
-                "eks:UpdateNodegroupConfig"
-            ],
-            "Effect": "Allow",
-            "Resource": "*"
-        },
-        {
-            "Action": [
-                "autoscaling:DescribeAutoScalingGroups",
-                "autoscaling:UpdateAutoScalingGroup"
-            ],
-            "Effect": "Allow",
-            "Resource": "*"
-        },
-        {
-            "Action": [
-                "sqs:GetQueueAttributes",
-                "sqs:GetQueueUrl"
-            ],
-            "Effect": "Allow",
-            "Resource": "*"
-        }
-    ]
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "eks:DescribeNodegroup",
+        "eks:UpdateNodegroupConfig"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Action": [
+        "autoscaling:DescribeAutoScalingGroups",
+        "autoscaling:UpdateAutoScalingGroup"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Action": [
+        "sqs:GetQueueAttributes",
+        "sqs:GetQueueUrl"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Action": [
+        "ec2:DescribeLaunchTemplates",
+        "ec2:CreateLaunchTemplate",
+        "ec2:CreateFleet",
+        "ec2:RunInstances",
+        "ec2:CreateTags",
+        "ec2:DescribeSubnets",
+        "eks:DescribeCluster",
+        "iam:GetRole",
+        "iam:CreateRole",
```

Comment:
I think this is a scary permission.  I think it would be better to accept a pre-created role ARN as a parameter somewhere, and don't create the Instance Role ourselves.

(Probably similar with some of the other permissions too, but this one jumps out immediately.)

Reply:
Yeah i was thinking about this. Here's how I was thinking about it:
For demo purposes, we can grant this permission and bootstrap the resources ourselves. For the production version, users create the resources themselves. The code already supports this if you want to remove these permissions and BYO Role or LaunchTemplate. WDYT?

Reply:
IMO, if this is for demo purposes we should move it to  [karpenter-aws-demo](https://github.com/ellistarn/karpenter-aws-demo/)
Here we should keep the minimum permissions we need, if we think we should never be creating profile/roles ourselves (if they don't exist), we should document and let users create them.

Reply:
I agree with you here. Let's add this to the list of design considerations. In the short term, I have some changes here.

Reply:
Yeah, 'demo' features have a tendency to leak out and become the 'default' install.  This one in particular seems like it's worth some effort to think about how we avoid it - even if the install instructions turn out to be two commands instead of one.